prow/github/client.go

```diff
@@ -1218,6 +1219,34 @@ func (c *client) ListIssueComments(org, repo string, number int) ([]IssueComment
 	return comments, nil
 }
 
+// ListIssues returns all open issues, including pull requests
+//
+// Each page of results consumes one API token.
+//
+// See https://developer.github.com/v3/issues/#list-issues-for-a-repository
+func (c *client) ListIssues(org, repo string) ([]Issue, error) {
+	c.log("ListIssues", org, repo)
+	if c.fake {
+		return nil, nil
+	}
+	path := fmt.Sprintf("/repos/%s/%s/issues", org, repo)
```

Comment:
Maybe we could implement the `since` parameter to limit the number of issues to return, and save API tokens in big repos like k/k...

Reply:
Yeah I think it's basically going to be required to run this as-is with ghproxy

Reply:
I like this idea for querying the issues, specifically.

If this sends a request, say, every hour, to k/k (~2100 issues, so ~21 tokens), then ghProxy will have to rerequest the entire 21-page resource if any issue has been added or updated in the past hour. Adding a "since" parameter might save us 20 tokens every full poll. An authorized user gets 5000 an hour, so I don't think this is necessary, but it's a nice-to-have.

We _shouldn't_ do this for the comments requests, though. Those API calls are pretty likely to be unchanged after an hour, and asking for the comments since a new timestamp is a new request, which consumes a new token. That's about 2100 tokens every hour; the real token-eater.

Reply:
I think this hinges also on my question about why this shouldn't be a plugin. If it is, we operate incrementally and only when something changes.